pkg/landscaper/installations/imports/conditional_imports_test.go

```diff
@@ -0,0 +1,192 @@
+// SPDX-FileCopyrightText: 2020 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package imports_test
+
+import (
+	"context"
+
+	"github.com/gardener/component-spec/bindings-go/ctf"
+	"github.com/go-logr/logr/testing"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	lsv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	"github.com/gardener/landscaper/pkg/api"
+	"github.com/gardener/landscaper/pkg/landscaper/installations"
+	"github.com/gardener/landscaper/pkg/landscaper/installations/imports"
+	"github.com/gardener/landscaper/pkg/landscaper/installations/subinstallations"
+	lsoperation "github.com/gardener/landscaper/pkg/landscaper/operation"
+	componentsregistry "github.com/gardener/landscaper/pkg/landscaper/registry/components"
+	kutil "github.com/gardener/landscaper/pkg/utils/kubernetes"
+	"github.com/gardener/landscaper/test/utils"
+	"github.com/gardener/landscaper/test/utils/envtest"
+)
+
+var _ = Describe("ConditionalImports", func() {
+
+	var (
+		op *installations.Operation
+
+		instRef types.NamespacedName
+		cmRef   lsv1alpha1.ObjectReference
+
+		fakeInstallations map[string]*lsv1alpha1.Installation
+		fakeClient        client.Client
+		fakeCompRepo      ctf.ComponentResolver
+	)
+
+	BeforeEach(func() {
+		var (
+			err   error
+			state *envtest.State
+		)
+
+		instRef = kutil.ObjectKey("conditional-import-inst", "test8")
+		cmRef = lsv1alpha1.ObjectReference{
+			Name:      "inst-import",
+			Namespace: instRef.Namespace,
+		}
+
+		fakeClient, state, err = envtest.NewFakeClientFromPath("./testdata/state")
```

Comment:
if the state is not needed you could also write
```suggestion
		fakeClient, _, err = envtest.NewFakeClientFromPath("./testdata/state")
```